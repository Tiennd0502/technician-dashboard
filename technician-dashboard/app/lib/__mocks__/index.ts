export * from './table';
export * from './categories';
export * from './services';
