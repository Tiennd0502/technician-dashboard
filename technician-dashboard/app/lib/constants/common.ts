export const LOGO_URL = '/images/technician.svg';

export const AVATAR_DEFAULT = '/images/empty-avatar.svg';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};

export const NO_DATA = 'No data found';
