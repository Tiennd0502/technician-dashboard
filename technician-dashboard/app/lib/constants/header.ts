import { AVATAR_DEFAULT } from '.';

export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
  title: 'Product & Services',
};
