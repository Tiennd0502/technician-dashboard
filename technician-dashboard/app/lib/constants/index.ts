export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
