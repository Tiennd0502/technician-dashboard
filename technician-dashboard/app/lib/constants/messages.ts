export const ERROR_MESSAGE = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required.`,
};
