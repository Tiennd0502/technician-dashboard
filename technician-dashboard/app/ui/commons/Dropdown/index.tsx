'use client';

import { memo } from 'react';
import { Avatar, Flex, Button, Text, Menu, MenuItem, MenuButton, MenuList } from '@chakra-ui/react';

import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons';

interface UserCardProps {
  avatar: string;
  name: string;
}

const Dropdown = ({ name, avatar }: UserCardProps) => (
  <Menu>
    {({ isOpen }) => (
      <>
        <MenuButton
          data-testid='btn-dropdown'
          as={Button}
          variant='ghost'
          rightIcon={
            isOpen ? (
              <ChevronUpIcon data-testid='up-icon' color='primary.500' />
            ) : (
              <ChevronDownIcon data-testid='down-icon' color='primary.500' />
            )
          }
        >
          <Flex alignItems='center' gap='3'>
            <Avatar name={name} src={avatar} aria-label='Avatar User' />
            <Text>{name}</Text>
          </Flex>
        </MenuButton>
        <MenuList data-testid='dropdown-options'>
          <MenuItem>Profile</MenuItem>
          <MenuItem>Change password</MenuItem>
        </MenuList>
      </>
    )}
  </Menu>
);

export default memo(Dropdown);
