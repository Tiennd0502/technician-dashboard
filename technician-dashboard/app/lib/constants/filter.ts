import { SORT_TYPE } from '@/lib/interfaces';

export const PRODUCT_LIMIT = 5;

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
