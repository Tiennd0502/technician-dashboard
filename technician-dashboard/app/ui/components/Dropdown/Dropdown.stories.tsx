import type { Meta, StoryObj } from '@storybook/react';

import { AVATAR_DEFAULT } from '@/lib/constants';

import Dropdown from '.';

const meta: Meta<typeof Dropdown> = {
  title: 'Components/Dropdown',
  component: Dropdown,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Dropdown>;

export const Default: Story = {
  args: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
};
