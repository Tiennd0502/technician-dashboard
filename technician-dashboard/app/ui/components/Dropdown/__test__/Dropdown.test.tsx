import { render } from '@testing-library/react';

import { AVATAR_DEFAULT } from '@/lib/constants';

import Dropdown from '../index';

describe('Dropdown test cases', () => {
  const props = {
    name: 'John',
    avatar: AVATAR_DEFAULT,
  };

  test('should render correctly', () => {
    const { container } = render(<Dropdown {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('get list options in dropdown component', () => {
    const { getByTestId } = render(<Dropdown {...props} />);

    const avatar = getByTestId('dropdown-options');
    expect(avatar).toBeTruthy();
  });
});
