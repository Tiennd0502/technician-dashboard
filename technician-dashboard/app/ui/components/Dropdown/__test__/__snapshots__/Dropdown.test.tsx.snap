// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown test cases should render correctly 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-4xx2wk"
    id="menu-button-:r1:"
    type="button"
  >
    <span
      class="css-xl71ch"
    >
      <div
        class="css-guqklo"
      >
        <span
          class="chakra-avatar css-1ruhx4w"
        >
          <div
            aria-label="John"
            class="chakra-avatar__initials css-0"
            role="img"
          >
            J
          </div>
        </span>
        <p
          class="chakra-text css-0"
        >
          John
        </p>
      </div>
    </span>
    <span
      class="chakra-button__icon css-1hzyiq5"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-q5208i"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      data-testid="dropdown-options"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="0"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Profile
      </button>
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="1"
        id="menu-list-:r1:-menuitem-:r3:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Change password
      </button>
    </div>
  </div>
</div>
`;
